src/core/auth/qgsauthmethod.h

```diff
@@ -190,6 +196,8 @@ class CORE_EXPORT QgsAuthMethod : public QObject
     QgsAuthMethod::Expansions mExpansions;
     QStringList mDataProviders;
     int mVersion = 0;
+    QMutex *mMutex;
```

Comment:
Could this be a plain variable rather than a pointer? Would make memory management a bit safer.

Reply:
@m-kuhn, it was a plain variable. But I could not find a way to force the recursive behavior with the default ctor. Do you know a way to do that?

Reply:
QgsAuthMethod::QgsAuthMethod
    : mMutex( QMutex::Recursive )

Reply:
thanks!

Reply:
np, sorry for the shorthand style, I'm a bit in a rush today ;)